net-vpn/openvpn: add 2.6.14 with many improvements

fkobi opened the pull request:
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

fkobi committed:
net-vpn/openvpn: improve metadata

- indent with tabs to reduce size
- rework longdescription
- improve mbedtls description
- add <bugs-to> & <changelog>

Signed-off-by: Filip Kobierski <fkobi@pm.me>

thesamesam commented on net-vpn/openvpn/openvpn-2.6.14.ebuild:
```diff
@@ -0,0 +1,160 @@
+
+	pushd tests/unit_tests &>/dev/null || die
+	elog "Running unit tests"
+	emake check
```
You can just do 
```suggestion
	emake -C tests/unit_tests check
```

thesamesam commented on net-vpn/openvpn/openvpn-2.6.14.ebuild:
```diff
@@ -0,0 +1,160 @@
+SLOT="0"
+IUSE="dco down-root examples inotify iproute2 +lz4 +lzo mbedtls pam pkcs11 +plugins selinux systemd test"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
```
Did you take into consideration ulm's feedback about REQUIRED_USE?

thesamesam commented on net-vpn/openvpn/openvpn-2.6.14.ebuild:
```diff
@@ -0,0 +1,160 @@
+REQUIRED_USE="
+	dco? ( !iproute2 )
+	pkcs11? ( !mbedtls )
+	!plugins? ( !inotify !pam !down-root )
```
I don't think this is the most helpful way of writing this for users to understand.

thesamesam commented on net-vpn/openvpn/openvpn-2.6.14.ebuild:
```diff
@@ -0,0 +1,160 @@
+
+COMMON_DEPEND="
+	sys-libs/libcap-ng:=
+	dco? ( >=net-vpn/ovpn-dco-0.2 >=dev-libs/libnl-3.2.29:= )
```
Usual style is:
```
dco? (
	a
	b
)
```

thesamesam commented:
> Also I think this is could get a higher priority due to https://bugs.gentoo.org/953118

Security bumps are best done without any huge refactoring.

fkobi committed:
net-vpn/openvpn: improve ebuild

Changes in top-down order:
- sorted inherits
- autoreconf conditionally
- fix LICENSE
  - BSD: COPYING/NSIS
  - GPL-2+: m4/pkg.m4
  - GPL-2+-openssl: lzo
- remove USE=openssl
- shorten REQUIRED_USE for plugins
- adjust dco? style
- group BDEPEND operations together
- add DOCS & drop unused file
- assign myeconfargs directly for simplicity
- treat pcks as a normal use flag
  - it's guarded by required use either way
- use usev instead of usex
- execute unit tests with -C
- add -type to find & change quote style
- use brace expansion for {up,down}.sh doexe
- capitalize first comment letter
- remove obsolete comments
- remove obsolete bug workaround
- remove stray space before ; in ifs

also sorted stuff

Signed-off-by: Filip Kobierski <fkobi@pm.me>

thesamesam reviewed (commented):
Please do squash with rationale added and then can look again. Looks OK so far.

thesamesam commented on net-vpn/openvpn/openvpn-2.6.14-r1.ebuild:
```diff
@@ -0,0 +1,165 @@
+# Copyright 1999-2025 Gentoo Authors
```
You've (again) not synced live.

fkobi committed:
net-vpn/openvpn: drop 2.6.12-r1

Bug: https://bugs.gentoo.org/953118
Signed-off-by: Filip Kobierski <fkobi@pm.me>

fkobi committed:
net-vpn/openvpn: sync live

Signed-off-by: Filip Kobierski <fkobi@pm.me>
